CMakeLists.txt

```diff
@@ -5,12 +5,24 @@ set(CMAKE_CXX_STANDARD 14)
 
 option(P2T_BUILD_TESTS "Build tests" OFF)
 option(P2T_BUILD_TESTBED "Build the testbed application" OFF)
+option(BUILD_SHARED_LIBS "Build as a shared library" OFF)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
```

Comment:
Hm ... are those three lines needed? I'm afraid that it might break something, especially when using exotic generators like Xcode with iOS.

Reply:
I've removed those lines in the latest version of the PR. The issue I faced, and the reason why I added them, was that on Windows when building poly2tri as a shared lib (DLL) the executables for the testbed and the unit tests would not find the library at runtime because it was generated in a different directory.

However, the testbed and unit tests build and run correctly with static linkage, which is the default. A user who wants to build poly2tri as a shared lib is not required to build the tesbed nor the unit test.